hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -63,9 +62,6 @@
   @Inject
   private OzoneConfiguration ozoneConfiguration;
 
-  @Inject
-  private Text omService;
-
   @Inject
   private String omServiceID;
```

Comment:
If we move the initialization of the `omServiceID` to `createOzoneClient` then we can delete `OzoneServiceProvider.java`

Reply:
The initialization should still effectively be Application Scoped as the client is initialized only once.